src/library/scala/collection/immutable/List.scala

```diff
@@ -405,6 +405,90 @@ sealed abstract class List[+A] extends AbstractSeq[A]
   // Create a proxy for Java serialization that allows us to avoid mutation
   // during de-serialization.  This is the Serialization Proxy Pattern.
   protected final def writeReplace(): AnyRef = new List.SerializationProxy(this)
+
+  override def filter(p: A => Boolean): List[A] = filterImpl(p, isFlipped = false)
```

Comment:
@som-snytt My type foo is not strong enough - I can't work out what the correct type signature here looks like - any hints greatly appreciated